user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -47,313 +141,151 @@ public class UserOperatorConfig {
-    private final int controllerThreadPoolSize;
-    private final long cacheRefresh;
-    private final int batchQueueSize;
-    private final int batchMaxBlockSize;
-    private final int batchMaxBlockTime;
-    private final int userOperationsThreadPoolSize;
+    public static final ConfigParameter<Boolean> ACLS_ADMIN_API_SUPPORTED = new ConfigParameter<>(STRIMZI_ACLS_ADMIN_API_SUPPORTED, BOOLEAN, "true");
+    /** Indicates whether KRaft is used in the Kafka cluster */
+    public static final ConfigParameter<Boolean> KRAFT_ENABLED = new ConfigParameter<>(STRIMZI_KRAFT_ENABLED, BOOLEAN, "false");
+    /** Timeout for internal operations specified in milliseconds */
+    public static final ConfigParameter<Long> OPERATION_TIMEOUT_MS = new ConfigParameter<>(STRIMZI_OPERATION_TIMEOUT_MS, LONG, "300000");
+    /** Indicates the size of the StrimziPodSetController work queue */
+    public static final ConfigParameter<Integer> WORK_QUEUE_SIZE = new ConfigParameter<>(STRIMZI_WORK_QUEUE_SIZE, INTEGER, "1024");
+    /** Size of the pool of the controller threads used to reconcile the users */
+    public static final ConfigParameter<Integer> CONTROLLER_THREAD_POOL_SIZE = new ConfigParameter<>(STRIMZI_CONTROLLER_THREAD_POOL_SIZE, INTEGER, "50");
+    /** Refresh interval for the cache storing the resources from the Kafka Admin API */
+    public static final ConfigParameter<Long> CACHE_REFRESH_INTERVAL_MS = new ConfigParameter<>(STRIMZI_CACHE_REFRESH_INTERVAL_MS, LONG, "15000");
+    /** Maximal queue for requests when micro-batching the Kafka Admin API requests */
+    public static final ConfigParameter<Integer> BATCH_QUEUE_SIZE = new ConfigParameter<>(STRIMZI_BATCH_QUEUE_SIZE, INTEGER, "1024");
+    /** Maximal batch size for micro-batching the Kafka Admin API requests */
+    public  static final ConfigParameter<Integer> BATCH_MAXIMUM_BLOCK_SIZE = new ConfigParameter<>(STRIMZI_BATCH_MAXIMUM_BLOCK_SIZE, INTEGER, "100");
+    /** Maximal batch time for micro-batching the Kafka Admin API requests */
+    public static final ConfigParameter<Integer> BATCH_MAXIMUM_BLOCK_TIME_MS = new ConfigParameter<>(STRIMZI_BATCH_MAXIMUM_BLOCK_TIME_MS, INTEGER, "100");
+    /** Size of the thread pool for user operations done by KafkaUserOperator and the classes used by it */
+    public static final ConfigParameter<Integer> USER_OPERATIONS_THREAD_POOL_SIZE = new ConfigParameter<>(STRIMZI_USER_OPERATIONS_THREAD_POOL_SIZE, INTEGER, "4");
+    /** Additional configuration for the Kafka Admin Client */
+    public static final ConfigParameter<String> KAFKA_ADMIN_CLIENT_CONFIGURATION = new ConfigParameter<>(STRIMZI_KAFKA_ADMIN_CLIENT_CONFIGURATION, STRING, false);
+    /** Lit of maintenance windows */
+    public static final ConfigParameter<String> MAINTENANCE_TIME_WINDOWS = new ConfigParameter<>(STRIMZI_MAINTENANCE_TIME_WINDOWS, STRING, false);
+
+    private static final Map<String, ConfigParameter<?>> CONFIG_VALUES = Map.ofEntries(
```

Comment:
This feels wrong. Having two places to update when adding a parameter is a bug waiting to happen. It should be possible, some how, to build the collection as a side-effect of defining a parameter.